### PR TITLE
Add cadence 1.0 functionality.

### PR DESCRIPTION
**Pull request by aishairzay**
The cadence 1.0 upgrades do NOT contain any changes for the UI to point to previewnet, all catalog addition requests, etc should be done with transactions for the time-being.

Also, the transaction generation portion of the catalog is not configured to work. This can be added in the future if it is a requested feature when all dependency contracts that are used by the tx generation are available with cadence 1.0 compatibility.

Also all JS testing is not currently being used for testing in favor of the new Cadence testing framework. It has not been deleted because there is an update coming to flow-js-testing which may allow the old testing to work in the future.

### CODE REVIEW

File: cadence/contracts/DapperUtilityCoin.cdc
Comment (joshuahannan) [inline cadence/contracts/DapperUtilityCoin.cdc:6]: This can be removed

File: cadence/contracts/DapperUtilityCoin.cdc
Comment (joshuahannan) [inline cadence/contracts/DapperUtilityCoin.cdc:52]: Looks like you're using an old version of the token standards here. The entitlement should be `FungibleToken.Withdraw`, not `Withdrawable`

File: cadence/contracts/DapperUtilityCoin.cdc
Comment (joshuahannan) [inline cadence/contracts/DapperUtilityCoin.cdc:73]: This can be deleted

File: cadence/contracts/MultipleNFT.cdc
Comment (joshuahannan) [inline cadence/contracts/MultipleNFT.cdc:1]: This file should be deleted. it is not part of the standard any more

File: cadence/contracts/NFTCatalog.cdc
Comment (joshuahannan) [inline cadence/contracts/NFTCatalog.cdc:84]: Should this method have an entitlement? It looks like it should only be callable by an admin

File: cadence/contracts/NFTCatalog.cdc
Comment (joshuahannan) [inline cadence/contracts/NFTCatalog.cdc:199]: Can this be `view`?

File: cadence/contracts/NFTCatalog.cdc
Comment (joshuahannan) [inline cadence/contracts/NFTCatalog.cdc:203]: `view`?

File: cadence/contracts/NFTCatalog.cdc
Comment (joshuahannan) [inline cadence/contracts/NFTCatalog.cdc:211]: `view`?

File: cadence/contracts/NFTCatalog.cdc
Comment (joshuahannan) [inline cadence/contracts/NFTCatalog.cdc:215]: `view`?

File: cadence/contracts/NFTCatalog.cdc
Comment (joshuahannan) [inline cadence/contracts/NFTCatalog.cdc:257]: `view`?

File: cadence/contracts/NFTCatalog.cdc
Comment (joshuahannan) [inline cadence/contracts/NFTCatalog.cdc:265]: `view`?

**Review by joshuahannan — COMMENTED** [review]
Make sure you have updated all your dependencies and have copied the latest versions of the standard contracts from their respective feature branches. It looks like your contracts are still relying on versions of the token standards that are outdated

File: cadence/contracts/DapperUtilityCoin.cdc
Comment (aishairzay) [reply cadence/contracts/DapperUtilityCoin.cdc:6]: Sorry yeah, the things relying on dapperutilitycoin in this repo were changed prior to the newest token standard updates (This may be true for a few other contracts too). None of the catalog actually relies on dapperutilitycoin aside from the TX generation which i'm not sure if we need to continue to support or not.

The main ones to review are the testing, nftcatalog, retrieval, + nftadmin contracts. The one that the flow.json deploys to previewnet, and the cadence testing goes through

File: cadence/scripts/get_all_nfts_and_views_in_account.cdc
Comment (joshuahannan) [inline cadence/scripts/get_all_nfts_and_views_in_account.cdc:29]: It looks like the only thing you use the account object for here is issuing a capability and publishing a capability. Not sure if you need all the other entitlements. This applies to other scripts and transactions that you've modified here too, so I would look through them and make sure you have the minimal set of entitlements on each one

File: cadence/scripts/get_nft_collections_for_nft_type.cdc
Comment (joshuahannan) [inline cadence/scripts/get_nft_collections_for_nft_type.cdc:1]: I'm seeing in some transactions you use the `import X from "X"` format for the imports, but in others you use `import "x"`. Probably would be good to standardize

File: cadence/transactions/add_to_nft_catalog.cdc
Comment (joshuahannan) [inline cadence/transactions/add_to_nft_catalog.cdc:17]: Same comment here about Account entitlements. It looks like all you are doing here is borrowing a reference from storage, so the only entitlement you need is `BorrowValue` I believe. You should go through all the transactions and make sure you are using only the minimal entitlements required for the transaction.

File: cadence/transactions/mint_example_nft.cdc
Comment (joshuahannan) [inline cadence/transactions/mint_example_nft.cdc:64]: I uncommented [this section in the flow-nft repo](https://github.com/onflow/flow-nft/blob/standard-v2/transactions/mint_nft.cdc), so you can copy that transaction and use it since it uses royalties properly now

**Review by joshuahannan — COMMENTED** [review]
Left a few more comments, but I'm gonna hold off on reviewing the rest until all the contracts and transactions have been updated to use the latest versions of the token standards and core contracts. Some of these look like they are a few months old. Let me know if you have any questions about which versions to use! They should be the latest versions in the feature branches in flow-ft, flow-nft, and flow-core-contracts

**Comment by aishairzay** [comment]
Thanks @joshuahannan, latest push should address everything mentioned and some more. I also removed a lot of code for areas of the catalog that are not relevant anymore with flix around (transaction generation).

**Review by joshuahannan — APPROVED** [review]

**Comment by joshuahannan** [comment]
@sisyphusSmiling I got the Cadence tests passing here and re-staged the contract on testnet so it would hopefully pass the next migration. I need help with all the JS stuff, so if you could take a look and try to get the tests passing at some point, that would be super helpful. There isn't a huge rush thought since we're already staged on testnet.

I do need to try to figure out where the previewnet private key information is to upgrade the contract there, so if you know anything about that, please let me know

**Comment by sisyphusSmiling** [comment]
No problem, @joshuahannan. I'll pick up JS tests later this week

**Review by sisyphusSmiling — COMMENTED** [review]
One callout re: entitlements in NFTCatalogAdmin

**Comment by jribbink** [comment]
> FYI I updated flow-js-testing to `0.6.0-stable-cadence.2` and flow-cadut to `^0.3.0-stable-cadence.1` plus some audit fixes. I wasn't able to get the js tests to pass and kept getting `thrown: undefined` on all the tests' `beforeEach` setup.

I'm in the process of pushing through some more fixes for JS Testing now.  It may solve the issues here.

**Comment by jribbink** [comment]
@sisyphusSmiling @joshuahannan There's a PR here https://github.com/onflow/nft-catalog/pull/152 for updating Flow JS Testing.  When I run it locally, it _seems_ to work, but it's failing tests because of issues in scripts/tx:

Screenshot 2024-05-22 at 2 53 52 PM

**Comment by joshuahannan** [comment]
@jribbink I'm not sure what the errors in the CI mean. They don't look like Cadence errors. Mostly seeing:
* `TypeError: Cannot read properties of undefined (reading 'kill')`
* `thrown: "Could not determine Flow CLI version, please make sure it is installed and available in your PATH"`

Not sure what I need to change about the JS tests to fix those.

I also can't run the tests locally yet. Still trying to figure out how to install all the correct dependencies

**Comment by jribbink** [comment]
> @jribbink I'm not sure what the errors in the CI mean. They don't look like Cadence errors. Mostly seeing:
> 
> * `TypeError: Cannot read properties of undefined (reading 'kill')`
> * `thrown: "Could not determine Flow CLI version, please make sure it is installed and available in your PATH"`
> 
> Not sure what I need to change about the JS tests to fix those.
> 

The CI issue on GitHub is coming from the fact that the install script changed for `flow-c1` recently, it just needs to be updated to the new script.

> I also can't run the tests locally yet. Still trying to figure out how to install all the correct dependencies. When I run `npm test`, i see this:
> 
> ```
>  FAIL  cadence/__test__/test/nftcatalog.test.js
>   ● Test suite failed to run
> 
>     Cannot find module '@onflow/flow-js-testing' from 'cadence/__test__/test/nftcatalog.test.js'
> ```

This issue just looks like you need to run `npm install` before trying to run the tests locally.

Re. `TypeError: Cannot read properties of undefined (reading 'kill')`, this will probably disappear when the other issues are solved, as it's most likely a symptom of the emulator not starting correctly/bad error reporting in Flow JS Testing

**Comment by joshuahannan** [comment]
Is it okay if I just merge this? Does it matter if the integration tests are completely passing or not since we're deprecating nft catalog anyway?

**Pull request merged**
